### DISCUSSION TITLE
File Upload

### DISCUSSION THREAD

**Discussion by cfuent-es** [livewire/livewire#5354] [1. Help]
Hello,

i'm having a weird problem using file upload. They work normally as far as I was able to test locally, but on the server, most of the time, the file does not pass the file validation, even though it is the same file that was previously accepted.

I removed the validations and the error on the server is that Storage cannot work with null in its parameter:

Call to a member function store() on null

See, this happens most of the time, but sometimes the file is saved normally. It's hard to know with random behavior like that.

I've seen a long thread ([3084](https://github.com/livewire/livewire/discussions/3084)), but the problem doesn't seem to be resolved. Could someone guide me? I'm using a sub-domain (listas.aromaearte.com.br).

Thanks.

**Comment by mrl22** [comment]
To rule it out, have you checked PHP upload_max_filesize and post_max_size? Try a file under 2MB.

**Reply by cfuent-es** [reply]
My file is 472kb, and as I said, it has already managed to send 11 files to the system, but it works whenever it wants...

**Comment by Martleby** [comment]
Have a check in your Laravel.log and see if you can find an exception logged when the upload fails.

We had a discussion #5180 where we noticed some behavior on a Windows server some file types wouldn't save - I suspect due to an invalid file name under Windows when the mime type can't be determined.

**Reply by cfuent-es** [reply]
The upload doesn't fail, it doesn't pass file validation. If I take the validation off, it shows the error that the store() function cannot save null. By the discussion that I put the link, there are people pointing a problem of Laravel for sub-domains, and another with problems in the behavior in NGinx. Really very confused.

**Reply by Martleby** [reply]
Still, I think you should check your Laravel log :)
